tests/test_discovery.py

```diff
@@ -223,8 +223,109 @@ def test_subactors_unregister_on_cancel_remote_daemon(
             spawn_and_check_registry,
             arb_addr,
             use_signal,
-            True,
+            True,  # remote arbiter
             with_streaming,
             # XXX: required to use remote daemon!
             arbiter_addr=arb_addr
         )
+
+
+async def streamer(agen):
+    async for item in agen:
+        print(item)
+
+
+async def close_chans_before_nursery(
+    arb_addr: tuple,
+    use_signal: bool,
+    remote_arbiter: bool = False,
+) -> None:
+
+    # logic for how many actors should still be
+    # in the registry at teardown.
+    if remote_arbiter:
+        entries_at_end = 2
+    else:
+        entries_at_end = 1
+
+    async with tractor.get_arbiter(*arb_addr) as aportal:
+        try:
+            get_reg = partial(aportal.run, 'self', 'get_registry')
+
+            async with tractor.open_nursery() as tn:
+                portal1 = await tn.run_in_actor('consumer1', stream_forever)
+                agen1 = await portal1.result()
+
+                portal2 = await tn.start_actor('consumer2', rpc_module_paths=[__name__])
+                agen2 = await portal2.run(__name__, 'stream_forever')
+
+                async with trio.open_nursery() as n:
+                    n.start_soon(streamer, agen1)
+                    n.start_soon(cancel, use_signal, .5)
+                    try:
+                        await streamer(agen2)
+                    finally:
+                        # Kill the root nursery thus resulting in
+                        # normal arbiter channel ops to fail during
+                        # teardown. It doesn't seem like this is
+                        # reliably triggered by an external SIGINT.
+                        # tractor.current_actor()._root_nursery.cancel_scope.cancel()
```

Comment:
Probably needs to get deferred to a new test module:
`tests_hard_kill.py` or something.

Reply:
Made #145.